client/evilwallet/wallets.go

```diff
@@ -73,28 +74,60 @@ func (w *Wallets) GetUnspentOutput(wallet *Wallet) *Output {
 	return wallet.GetUnspentOutput()
 }
 
-func (w *Wallets) FreshWallet() *Wallet {
-	if len(w.faucetWallets) > 0 {
-		return w.wallets[w.faucetWallets[0]]
+// IsFaucetWalletAvailable checks if there is any faucet wallet left.
+func (w *Wallets) IsFaucetWalletAvailable() bool {
+	return len(w.faucetWallets) > 0
+}
+
+// FreshWallet returns the first non-empty wallet from the faucetWallets queue. If current wallet is empty,
+// it is removed and the next enqueued one is returned.
+func (w *Wallets) FreshWallet() (*Wallet, error) {
+	if !w.IsFaucetWalletAvailable() {
+		return nil, errors.New("no faucet wallets available, need to request more funds")
 	}
-	return nil
+	wallet := w.wallets[w.faucetWallets[0]]
+	if wallet.IsEmpty() {
+		w.removeWallet(fresh)
+		if !w.IsFaucetWalletAvailable() {
+			return nil, errors.New("no faucet wallets available, need to request more funds")
+		}
+		// take next wallet
+		wallet = w.wallets[w.faucetWallets[0]]
+		if wallet.IsEmpty() {
+			return nil, errors.New("wallet is empty, need to request more funds")
+		}
+	}
+	return wallet, nil
 }
 
-func (w *Wallets) RemoveWallet() *Wallet {
+// removeWallet removes wallet, for fresh wallet it will be the first wallet in a queue.
+func (w *Wallets) removeWallet(wType WalletType) {
+	switch wType {
+	case fresh:
```

Comment:
What about `reuse` type?

Reply:
The `reuse` wallets are not implemented for now, I explined in the further comment.